File: JavaScript/958. Check Completeness of a Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function(root) {
    var q = new Array();
    var n;
    var f = false;
    q.push(root);
    while(q.length>0){
        n = q.shift();
        if(n == null){
            f = true;
        }
        else{
            if(f == true){
                return false;
            }
            q.push(n.left);
            q.push(n.right);
        }
    }
    return true;
};

File: Java/958. Check Completeness of a Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isCompleteTree(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        TreeNode n;
        boolean f = false;
        q.offer(root);
        while(q.size()>0){
            n = q.poll();
            if(n == null){
                f = true;
            }
            else{
                if(f == true){
                    return false;
                }
                q.offer(n.left);
                q.offer(n.right);
            }
        }
        return true;
    }
}
